### DISCUSSION TITLE
How to return a byte array as a Python `bytes` object?

### DISCUSSION THREAD

**Discussion by nurpax** [wjakob/nanobind#539] [Q&A]
I'm creating bindings that'd need to return some arrays of arbitrary byte buffers (uint8_t arrays) as a Python `bytes` object.  But I can't quite figure out where to start.

One alternative would be to return an ndarray instead, but the bindings are used in a context that otherwise expects to see `bytes` objects.  The `nb::bytes()` constructors seem to only support constructing from C strings but.

**Comment by wjakob** [comment]
Good catch, this is now fixed in 18376ff56d35c3adf98bc7ba77799fe6ca33a20b
